lib/absinthe/middleware/telemetry.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Absinthe.Middleware.Telemetry do
+  @moduledoc """
+  Gather and report telemetry about an individual resolver function
+  """
+  @telemetry_event [:absinthe, :resolver]
+
+  @behaviour Absinthe.Middleware
+
+  @impl Absinthe.Middleware
+  def call(%{middleware: [{{Absinthe.Resolution, :call}, resolver_fun} | _]} = resolution, _) do
```

Comment:
@binaryseed It occurs to me that pattern matching on the resolution middleware being the first may be a problem. Consider:
```
field :foo, :foo do
  middleware Auth # or NodeId.Parser or w/e
  resolve &foo/1
end
```

Perhaps instead of `resolver_fun` you really just want the entire rest of the middleware stack?

Reply:
Yeah, the issue I linked to above (https://github.com/binaryseed/new_relic_absinthe/pull/6) points this out too..

Reply:
Sure, in that PR though it's extracting just the resolver fun. I'm suggesting pushing the whole stack through Telemetry, and then if a telemetry handler only wants the resolver fun they can extract it from that list. The whole middleware stack is important though because all of the middleware can do IO and take time.

Reply:
OK, I think I get what you are suggesting now.. updated!

Reply:
perfect, thanks! This seems good to go from my view, am I missing anything?